plugins/woocommerce/tests/php/src/Internal/Admin/Settings/PaymentProviders/WooPayments/WooPaymentsServiceTest.php

```diff
@@ -0,0 +1,4903 @@
+							WooPaymentsService::ONBOARDING_STEP_STATUS_COMPLETED => time(),
+						),
+					),
+					WooPaymentsService::ONBOARDING_STEP_WPCOM_CONNECTION   => array(
+						'statuses' => array(
+							WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED => time() - 10,
+							WooPaymentsService::ONBOARDING_STEP_STATUS_COMPLETED => time(),
+						),
+					),
+					WooPaymentsService::ONBOARDING_STEP_TEST_ACCOUNT       => array(
+						'statuses' => array(
+							WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED => time() - 10,
+							WooPaymentsService::ONBOARDING_STEP_STATUS_COMPLETED => time(),
+						),
+					),
+					WooPaymentsService::ONBOARDING_STEP_BUSINESS_VERIFICATION => array(
+						'statuses' => array(
+							WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED => time() - 10,
+							WooPaymentsService::ONBOARDING_STEP_STATUS_COMPLETED => time(),
+						),
+					),
+				),
+				$default_recommended_pms,
+				$expected_pms_state,
+				$default_wpcom_connection,
+				$expected_wpcom_connection_state,
+				array(
+					'has_account'       => true,
+					'has_valid_account' => true,
+					'test_account'      => false,
+				),
+			),
+			'stored statuses (mixed)' => array(
+				array(
+					WooPaymentsService::ONBOARDING_STEP_PAYMENT_METHODS => WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED,
+					WooPaymentsService::ONBOARDING_STEP_WPCOM_CONNECTION   => WooPaymentsService::ONBOARDING_STEP_STATUS_NOT_STARTED,
+					// Forced due to the bad WPCOM connection status (requirements).
+					WooPaymentsService::ONBOARDING_STEP_TEST_ACCOUNT       => WooPaymentsService::ONBOARDING_STEP_STATUS_NOT_STARTED,
+					// Forced due to the bad WPCOM connection status (requirements).
+					WooPaymentsService::ONBOARDING_STEP_BUSINESS_VERIFICATION => WooPaymentsService::ONBOARDING_STEP_STATUS_NOT_STARTED,
+				),
+				array(
+					WooPaymentsService::ONBOARDING_STEP_PAYMENT_METHODS => array(
+						'statuses' => array(
+							WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED => time() - 10,
+						),
+					),
+					// Nothing about the WPCOM connection.
+					WooPaymentsService::ONBOARDING_STEP_TEST_ACCOUNT       => array(
+						'statuses' => array(
+							WooPaymentsService::ONBOARDING_STEP_STATUS_COMPLETED => time(),
+						),
+					),
+					WooPaymentsService::ONBOARDING_STEP_BUSINESS_VERIFICATION       => array(
+						'statuses' => array(
+							WooPaymentsService::ONBOARDING_STEP_STATUS_COMPLETED => time(),
+						),
+					),
+				),
+				$default_recommended_pms,
+				$expected_pms_state,
+				$default_wpcom_connection,
+				$expected_wpcom_connection_state,
+				$default_account_state,
+			),
+			'no stored statuses - WPCOM connection: store_connected, no connected owner' => array(
+				array(
+					WooPaymentsService::ONBOARDING_STEP_PAYMENT_METHODS => WooPaymentsService::ONBOARDING_STEP_STATUS_NOT_STARTED,
+					WooPaymentsService::ONBOARDING_STEP_WPCOM_CONNECTION   => WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED,
```

Comment:
how it will look like in UI?

Reply:
The user should be prompted to connect to WPCOM since the authorization is not there.

Reply:
Gotcha